plugins/completion/cmp/sources/sources.nix

```diff
@@ -1,49 +1,185 @@
+_:
 # Maps each source name to the plugin that provides it
 # TODO: make this a _list_ of mkCmpSourcePlugin arguments, inspired by lsp/language-servers
-{
-  async_path = "cmp-async-path";
-  buffer = "cmp-buffer";
-  calc = "cmp-calc";
-  dap = "cmp-dap";
-  cmdline = "cmp-cmdline";
-  cmp_ai = "cmp-ai";
-  cmp-clippy = "cmp-clippy";
-  cmp-cmdline-history = "cmp-cmdline-history";
-  cmp_pandoc = "cmp-pandoc-nvim";
-  cmp_tabby = "cmp-tabby";
-  cmp_tabnine = "cmp-tabnine";
-  codeium = "codeium-nvim";
-  conventionalcommits = "cmp-conventionalcommits";
-  copilot = "copilot-cmp";
-  crates = "crates-nvim";
-  dictionary = "cmp-dictionary";
-  digraphs = "cmp-digraphs";
-  emoji = "cmp-emoji";
-  fish = "cmp-fish";
-  fuzzy_buffer = "cmp-fuzzy-buffer";
-  fuzzy_path = "cmp-fuzzy-path";
-  git = "cmp-git";
-  greek = "cmp-greek";
-  latex_symbols = "cmp-latex-symbols";
-  look = "cmp-look";
-  luasnip = "cmp_luasnip";
-  nvim_lsp = "cmp-nvim-lsp";
-  nvim_lsp_document_symbol = "cmp-nvim-lsp-document-symbol";
-  nvim_lsp_signature_help = "cmp-nvim-lsp-signature-help";
-  nvim_lua = "cmp-nvim-lua";
-  npm = "cmp-npm";
-  omni = "cmp-omni";
-  pandoc_references = "cmp-pandoc-references";
-  path = "cmp-path";
-  rg = "cmp-rg";
-  snippy = "cmp-snippy";
-  spell = "cmp-spell";
-  tmux = "cmp-tmux";
-  treesitter = "cmp-treesitter";
-  ultisnips = "cmp-nvim-ultisnips";
-  vim_lsp = "cmp-vim-lsp";
-  vimwiki-tags = "cmp-vimwiki-tags";
-  vsnip = "cmp-vsnip";
-  yanky = "cmp_yanky";
-  zsh = "cmp-zsh";
-}
+[
+  {
+    name = "cmp-async-path";
```

Comment:
Definitely `pluginName`. At least I feel it would severely enhance clarity.

Reply:
I've done this, but I'm not convinced it is actually better. In the list it makes sense, but in `mkCmpSourcePlugin` it does not.

Remember `sources` is supposed to be a list of `mkCmpSourcePlugin` arguments, and other than a few special args, they get passed through to `mkVimPlugin`.